### ISSUE TITLE
Stop Button Doesn't Work

### ISSUE THREAD

**Issue by TannerRogalsky** [Ulydev/lovefiddle#1]
Currently, the stop button does not stop execution of the running code. I think you can fully halt execution of the code using this snippet:

``` javascript
try {
  Browser.mainLoop.pause();
  Browser.mainLoop.func = null;
  Module['noExitRuntime'] = false;
  Module['exit'](0);
} catch (e) {
  console.log(e);
}
```

But, as per [this discussion](https://groups.google.com/forum/#!topic/emscripten-discuss/NBWEtAHv760), there are some global initializers that will prevent just invoking `callMain` again. Gonna need to wrap that global state in a function so that it get recreated.

**Comment by Ulydev** [comment]
Thanks, I'm going to look into it!

**Comment by Ulydev** [comment]
Fixed in eeeb7e3dcb2be03248a06f4031219fca0164a2b6

**Comment by Ulydev** [comment]
It's very hacky, but it works. Might want to make this cleaner in a future release.
